### PR TITLE
Feature/add grader

### LINKED ISSUE #36: Add the Grader role to the form and template
See #33 

Add a role selector to the form so faculty can choose the role they wish to use for the student->course association.  This will require a template change and a change to the form object's logic.  The form object should store the chosen role.  Another ticket will be filed for acting on the data in that field.

### PR DESCRIPTION
**Pull request by lsat12357**
fixes #36

### CODE REVIEW

File: src/roles/roles.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package roles
+
+// Roles lists the roles
```
Comment (jechols) [inline src/roles/roles.go]: This comment should be a little clearer as to its intent: it's a list of all valid roles, or it's our full list of known role we support, or something like that.  Nitpicky, I know, but I want our code to be better than my first stab at this app.

File: src/roles/roles.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+var Roles = []string{"GE", "Grader"}
+
+func validateRole(role string) bool {
```
Comment (jechols) [inline src/roles/roles.go]: This will need to be capitalized and documented, thanks to Go's magic public vs. private logic.  As it is right now, because it's lowercase, it isn't accessible outside the "roles" package.

Also, a best practice: keep the word "Role" out of functions.  The code would be called via `roles.ValidateRole(x)` if the function name is kept as-is.  A succinct name like `IsValid` or even just `Valid` would be clearer to read: `if roles.Valid("foo") { ... }`

Comment (jechols) [reply src/roles/roles.go]: As an add-on the that second bit, you might be tempted to change the variable above from "Roles" to "All" or something in order to keep the word "Role" out.  That's a case that's a lot more subjective, it seems like.  There is a lot of code that repeats the package name because it's necessary and there just isn't a great way to describe the variable or function any other way.  This is rarely done with functions, hence my comment above, but it does happen with types and variables.

tl;dr: keep the variable named "Roles"

File: src/sci-server/form.go [hunk 3]
```diff
@@ -32,14 +33,17 @@ func (r *responder) getForm() (f *form, err error) {
...
+	f.Role = r.req.PostFormValue("role")
 
 	if f.DuckID == "" {
 		f.errors = append(f.errors, errors.New("duckid must be filled out"))
 	}
 	if f.CRN == "" {
 		f.errors = append(f.errors, errors.New("a course must be chosen"))
 	}
-
+	if f.Role == "" {
```
Comment (jechols) [inline src/sci-server/form.go]: This is semi-out-of-scope, but since you have the validation code already, it's probably worth adding the check in here rather than just checking if the value is blank.  The error message may need a tweak, but otherwise it'll be a pretty simple change, and knocks out another of our mini-tasks.
